x-pack/plugins/security_solution/public/explore/hosts/pages/details/index.tsx

```diff
@@ -167,6 +174,14 @@ const HostDetailsComponent: React.FC<HostDetailsProps> = ({ detailName, hostDeta
     [detailName]
   );
 
+  const entity = useMemo(() => ({ type: 'host' as const, name: detailName }), [detailName]);
+  const privileges = useAssetCriticalityPrivileges(entity.name);
```

Comment:
Error handling is missing when requesting the API for the privilege check. The user may not be able to access the `assetCriticalityPrivilegesRoute` API route itself (serverless _essentials_ tier). There are two options to solve that:

a) Add error handling to the API request and return empty privileges when some error happens. 

b) Remove the `access:${APP_ID}-entity-analytics` tag from the _privileges_ API route. And implement the checks for `appFeatures.advancedInsights` and _hasPrivileges_ for `${APP_ID}-entity-analytics` explicitly. 

(To reproduce that, disable `appFeature.advancedInsights` and login using a "non-superuser" user)

Reply:
Tested with Pablo, the error is implicitly handled by _useQuery_ from _react-query_, then `privileges.data` is _undefined_, so the user won't have permission and no error will be displayed. We are good.

Reply:
I tested this scenario, and `useQuery` does the error handling by default. 
It returns data as undefined when there is an error, and we consider data as optional when checking privileges. Like this `privileges.data?.has_...`



![Screenshot 2024-02-12 at 11 41 25](https://github.com/elastic/kibana/assets/1490444/f3011668-c87e-41a1-97e1-44e236f468d3)

![Screenshot 2024-02-12 at 11 46 44](https://github.com/elastic/kibana/assets/1490444/b3051bf2-7c2b-43e2-bbb7-3996d7282ede)

Reply:
@tiansivive @hop-dev I believe this comment raises an interesting question. 

> Should APIs that check privileges be public?

If we make the API public we would have to add a couple of checks to the implementation. 

I think it would be cleaner to treat all authorization errors the same way. The current implementation doesn't differentiate between HTTP 403 and other HTTP statuses. We could swallow real errors.

Reply:
Sorry I am missing the connection here, why would making the API public linked to handling authorisation errors?

Reply:
That sentence was confusing. By public, I meant removing `access:${APP_ID}-entity-analytics` privileges from the API:

https://github.com/machadoum/kibana/blob/fd30f2555f6a72902e5124baac36ed493e8bac41/x-pack/plugins/security_solution/server/lib/entity_analytics/asset_criticality/routes/privileges.ts#L26

Reply:
So is this right... the `access:${APP_ID}-entity-analytics` tag on the privileges API route means the user must have `appFeatures.advancedInsights` and `${APP_ID}-entity-analytics` to be able to call the route. 

But they do not need those privileges to see the hosts page ,so there could be a situation where the user navigates to the page and gets a 403 from the asset criticality privileges API. 

Is there a situation where that user then does have the necessary index permissions to use risk engine/asset criticality? 

If we made it public I think we would have to add fields to the response which says which which features/apps they require access to, in the same way we tell them which index privileges they are missing.

Reply:
Having the API route gated by the `entity-analytics` feature is not wrong, but it means the UI should not call this API if the "entity analytics" feature is not enabled. 
So, we could just add a check on the UI side, making sure the capability is enabled (I think it is `capabilities.siem.['entity-analytics']`) before doing the privileges API call, otherwise it means entity analytics is disabled as a whole, there's no need to do it.

Reply:
Ah OK, sorry I was complicating things. Yeah lets do it, @machadoum you happy to create an issue?

Reply:
Yes, I did not think about this solution at first. The two options I suggested in the first comment were indeed more complex. I think (pre) checking UI capabilities is easier and cleaner 👍

Reply:
I created a ticket that describes all the checks we should do before calling the privileges API. Here it is: https://github.com/elastic/security-team/issues/8689